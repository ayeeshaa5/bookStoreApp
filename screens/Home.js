import { StatusBar } from 'expo-status-bar';
import { ImageBackground, StyleSheet, Text, Button, Image, TouchableOpacity, View } from 'react-native';

const Home = ({ navigation }) => {

  const goToLogin = () => {
    navigation.navigate('Login');
  };
  const goToSignup = () => {
    navigation.navigate('Signup');
  };

  return (
    <View style={styles.container}>
      <ImageBackground source={require('../assets/images/bg7.png')} style={styles.image}>
        <Text style={styles.text1}>
          WELCOME TO EBOOKSTORE
        </Text>

        <TouchableOpacity testID='login-button' style={styles.Login} onPress={goToLogin}>
          <View style={styles.Loginbutton}><Text style={styles.Logintext}>Login</Text></View>
        </TouchableOpacity>

        <TouchableOpacity testID='signup-button' style={styles.Signup} onPress={goToSignup}>
          <View style={styles.Signupbutton}><Text style={styles.Signuptext}>Sign up</Text></View>
        </TouchableOpacity>
      </ImageBackground>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  image: {
    flex: 1,
    resizeMode: 'cover',
    justifyContent: 'center',
  },
  text1: {
    fontSize: 33,
    fontWeight: 'bold',
    fontStyle: 'italic',
    marginTop: '90%',
    paddingLeft: 20,
    color: 'black',
    borderColor: '#bdb76b',
  },
  Loginbutton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'white',
    borderRadius: 12,
    height: 40
  },
  Login: {
    width: 300,
    marginLeft: 50,
    marginTop: 50,
    borderColor: '#bdb76b',
    borderWidth: 3,
    borderRadius: 12,
  },
  Logintext: {
    fontSize: 19,
    color: 'black',
    fontWeight: 'bold',
  },
  Signupbutton: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'white',
    borderRadius: 12,
    borderColor:'#bdb76b',
    height: 40,
  },
  Signup: {
    width: 300,
    marginLeft: 50,
    marginTop: 10,
    borderColor: '#bdb76b',
    borderWidth: 3,
    borderRadius: 12,
  },
  Signuptext: {
    fontSize: 19,
    color: 'black',
    fontWeight: 'bold',
  }
});

export default Home;
