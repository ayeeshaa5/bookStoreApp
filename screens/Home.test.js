import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import Home from './Home';

test('renders Login and Signup buttons', () => {
  const mockNavigation = {
    navigate: jest.fn(),
  };

  const { getByTestId } = render(<Home navigation={mockNavigation} />);
  
  // Find the Login and Signup buttons
  const loginButton = getByTestId('login-button');
  const signupButton = getByTestId('signup-button');
  
  // Assert that the buttons are present
  expect(loginButton).toBeTruthy();
  expect(signupButton).toBeTruthy();
});


test('navigates to Login screen when Login button is pressed', () => {
  const mockNavigation = {
    navigate: jest.fn(),
  };

  const { getByTestId } = render(<Home navigation={mockNavigation} />);
  
  // Find the Login button
  const loginButton = getByTestId('login-button');
  
  // Simulate clicking the Login button
  fireEvent.press(loginButton);
  
  // Check if the navigate function is called with the correct screen name
  expect(mockNavigation.navigate).toHaveBeenCalledWith('Login');
});

test('navigates to Signup screen when Signup button is pressed', () => {
  const mockNavigation = {
    navigate: jest.fn(),
  };

  const { getByTestId } = render(<Home navigation={mockNavigation} />);
  
  // Find the Signup button
  const signupButton = getByTestId('signup-button');
  
  // Simulate clicking the Signup button
  fireEvent.press(signupButton);
  
  // Check if the navigate function is called with the correct screen name
  expect(mockNavigation.navigate).toHaveBeenCalledWith('Signup');
});
it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
