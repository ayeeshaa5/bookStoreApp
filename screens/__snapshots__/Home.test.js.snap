// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={1}
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        {
          "borderColor": "#bdb76b",
          "color": "black",
          "fontSize": 33,
          "fontStyle": "italic",
          "fontWeight": "bold",
          "marginTop": "90%",
          "paddingLeft": 20,
        }
      }
    >
      WELCOME TO EBOOKSTORE
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "#bdb76b",
          "borderRadius": 12,
          "borderWidth": 3,
          "marginLeft": 50,
          "marginTop": 50,
          "opacity": 1,
          "width": 300,
        }
      }
      testID="login-button"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 12,
            "height": 40,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 19,
              "fontWeight": "bold",
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderColor": "#bdb76b",
          "borderRadius": 12,
          "borderWidth": 3,
          "marginLeft": 50,
          "marginTop": 10,
          "opacity": 1,
          "width": 300,
        }
      }
      testID="signup-button"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "#bdb76b",
            "borderRadius": 12,
            "height": 40,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            {
              "color": "black",
              "fontSize": 19,
              "fontWeight": "bold",
            }
          }
        >
          Sign up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
